### PR TITLE
Fix iOS modal flicker on new architecture

### PR DESCRIPTION
**Pull request by skostilainen**
## Summary:

With new architecture modals with no animation flickers on top-left corner on iOS.

## Changelog:

[IOS] [FIXED] - use synchronous thread when updating modal bounds in modal host component view for more immediate updates

## Test Plan:

Create new react-native app with expo. Enable new architecture and take a copy of react-native example documentation of using "Modal" component. When opening the modal, notice the modal flickering on top-left corner. Use this PR to verify it's fixed.

- Before 

https://github.com/user-attachments/assets/e9a4d34a-48fd-4e67-9cff-9332f53b26ec

- After

https://github.com/user-attachments/assets/d6b6c5eb-3be9-4bd7-92f3-3f949e43ad03

### CODE REVIEW

File: packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm [hunk 1]
```diff
@@ -218,7 +218,7 @@ - (void)boundsDidChange:(CGRect)newBounds
 
   if (_state != nullptr) {
     auto newState = ModalHostViewState{RCTSizeFromCGSize(newBounds.size)};
-    _state->updateState(std::move(newState));
+    _state->updateState(std::move(newState), EventPriority::Sync);
```
Comment (sammy-SC) [inline packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm:221]: I don't think his API is available. Were you able to compile this on the main brach?

**Comment by sammy-SC** [comment]
The reason why this is happening is because Modal on iOS needs to be mounted to supply measurements of the screen.
Once it has that, it triggers layout and second paint.

One way to work around it is to wait until state [here](https://github.com/facebook/react-native/blob/main/packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm#L264) has non-zero screenSize and only then make the contents of Modal visible to the user. 
The content will be delayed by a few frames but in this scenario it is ok.

File: packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm (hunk 1 above)
Comment (skostilainen) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm:221]: It should be available in ConcreteState.h. Uses `EventPriority::AsynchronousBatched)` by default

**Comment by skostilainen** [comment]
Thanks for the comments. Great if it has been fixed in the RC branch already, I only tested against latest release

**Comment by skostilainen** [comment]
Fixed already by https://github.com/facebook/react-native/commit/14de1b708a637313f34aa0806e5c6f0ea6bb7417

**Pull request closed without merging**